clusterprovider/kubeflex/kubeflex.go

```diff
@@ -0,0 +1,208 @@
+	}
+
+	return err
+}
+
+// TODO: switch from CLI to getting this info via kube directives
+func (k KflexClusterProvider) Delete(name string, opts clusterprovider.Options) error {
+	logger := klog.Background()
+	logger.V(2).Info("Deleting KubeFlex cluster", "name", name)
+	cmd := exec.Command("kflex", "delete", name)
+	return cmd.Run()
+}
+
+// ListClusterNames: returns a list of clusters in KubeFlex that are
+// in the Ready condition.
+// TODO: switch from CLI to getting this info via kube directives
+func (k KflexClusterProvider) ListClustersNames() ([]string, error) {
+	var out bytes.Buffer
+	var listClusterNames []string
+	cmd := exec.Command("kubectl", "--context", "kind-kubeflex", "get", "controlplanes", "--no-headers")
+	cmd.Stdout = &out
+	err := cmd.Run()
+	if err != nil {
+		return nil, err
+	}
+	b := out.Bytes()
+	list := strings.Split(string(b), "\n")
+	for _, v := range list {
+		words := strings.Fields(v)
+		if len(words) > 2 && words[2] == "True" {
+			listClusterNames = append(listClusterNames, words[0])
+		}
+	}
+	return listClusterNames, nil
+}
+
+// Get: obtains the kubeconfig for the given lcName cluster.
+// TODO: switch from cli to kube directives
+func (k KflexClusterProvider) Get(lcName string) (clusterprovider.LogicalClusterInfo, error) {
+	cmd := "kubectl --context kind-kubeflex get secrets -n lc3-system admin-kubeconfig -o jsonpath='{.data.*}' | base64 -d"
+	cfg, err := exec.Command("bash", "-c", cmd).Output()
+	if err != nil {
+		return clusterprovider.LogicalClusterInfo{}, err
+	}
+	lcInfo := clusterprovider.LogicalClusterInfo{
+		Name:   lcName,
+		Config: string(cfg),
+	}
+	return lcInfo, nil
+}
+
+func (k KflexClusterProvider) ListClusters() ([]clusterprovider.LogicalClusterInfo, error) {
+	logger := klog.Background()
+	lcNames, _ := k.ListClustersNames()
+
+	lcInfoList := make([]clusterprovider.LogicalClusterInfo, 0, len(lcNames))
+
+	for _, lcName := range lcNames {
+		// TODO: switch from cli to kube directives
+		cmd := "kubectl --context kind-kubeflex get secrets -n lc3-system admin-kubeconfig -o jsonpath='{.data.*}' | base64 -d"
+		cfg, err := exec.Command("bash", "-c", cmd).Output()
+		if err != nil {
+			logger.Error(err, "Failed to fetch config for cluster", "name", lcName)
+		}
+
+		lcInfoList = append(lcInfoList, clusterprovider.LogicalClusterInfo{
+			Name:   lcName,
+			Config: string(cfg),
+		})
+	}
+
+	return lcInfoList, nil
+}
+
+func (k KflexClusterProvider) Watch() (clusterprovider.Watcher, error) {
+	w := &KflexWatcher{
+		ch:       make(chan clusterprovider.WatchEvent),
+		provider: &k}
+	k.watch = w
+	return w, nil
+}
+
+type KflexWatcher struct {
+	init     sync.Once
+	wg       sync.WaitGroup
+	ch       chan clusterprovider.WatchEvent
+	cancel   context.CancelFunc
+	provider *KflexClusterProvider
+}
+
+func (k *KflexWatcher) Stop() {
+	if k.cancel != nil {
+		k.cancel()
+	}
+	k.wg.Wait()
+	close(k.ch)
+}
+
+func (k *KflexWatcher) ResultChan() <-chan clusterprovider.WatchEvent {
```

Comment:
This seems like a generic implementation for non watchable providers. Maybe add this to our library as a special  ResultChan implementation that providers can just use instead of re-implement each time ?

Reply:
You are absolutely right.  This is a poor man's watcher.  Only question is whether we should do it now, or do it as part of the change to REST.

Reply:
Created issue #819 to track this.